Add 21 autogenerated filesets from rsa2elk devices

adriansr opened the pull request:
## What does this PR do?

This adds the following **experimental** filesets based on Apache 2 license device parsers:
- tomcat.log (from [apachetomcat](https://github.com/blookot/rsa2elk/tree/master/devices/apachetomcat))
- netscout.sightline (from [arborpeakflowsp](https://github.com/blookot/rsa2elk/tree/master/devices/arborpeakflowsp))
- barracuda.waf (from [barracudawaf](https://github.com/blookot/rsa2elk/tree/master/devices/barracudawaf))
- f5.bigipapm (from [bigipapm](https://github.com/blookot/rsa2elk/tree/master/devices/bigipapm))
- bluecoat.director (from [bluecoatdirector](https://github.com/blookot/rsa2elk/tree/master/devices/bluecoatdirector))
- cisco.nexus (from [cisconxos](https://github.com/blookot/rsa2elk/tree/master/devices/cisconxos))
- citrix.virtualapps (from [citrixxa](https://github.com/blookot/rsa2elk/tree/master/devices/citrixxa))
- cylance.protect (from [cylance](https://github.com/blookot/rsa2elk/tree/master/devices/cylance))
- f5.firepass (from [firepass](https://github.com/blookot/rsa2elk/tree/master/devices/firepass))
- fortinet.clientendpoint (from [forticlientendpoint](https://github.com/blookot/rsa2elk/tree/master/devices/forticlientendpoint))
- imperva.securesphere (from [impervawaf](https://github.com/blookot/rsa2elk/tree/master/devices/impervawaf))
- infoblox.nios (from [infobloxnios](https://github.com/blookot/rsa2elk/tree/master/devices/infobloxnios))
- juniper.junos (from [junosrouter](https://github.com/blookot/rsa2elk/tree/master/devices/junosrouter))
- kaspersky.av (from [kasperskyav](https://github.com/blookot/rsa2elk/tree/master/devices/kasperskyav))
- microsoft.dhcp (from [msdhcp](https://github.com/blookot/rsa2elk/tree/master/devices/msdhcp))
- tenable.nessus_security (from [nessusvs](https://github.com/blookot/rsa2elk/tree/master/devices/nessusvs))
- rapid7.nexpose (from [nexpose](https://github.com/blookot/rsa2elk/tree/master/devices/nexpose))
- radware.defensepro (from [radwaredp](https://github.com/blookot/rsa2elk/tree/master/devices/radwaredp))
- sonicwall.firewall (from [sonicwall](https://github.com/blookot/rsa2elk/tree/master/devices/sonicwall))
- squid.log (from [squid](https://github.com/blookot/rsa2elk/tree/master/devices/squid))
- zscaler.zia (from [zscalernss](https://github.com/blookot/rsa2elk/tree/master/devices/zscalernss))

## Why is it important?

This is an effort to generate as many as possible experimental input sources from a set of 300 Apache2-licensed log parsers.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Notes to reviewers

The modules in this PR are autogenerated. All follow the same format:
```
sonicwall/
├── README.md
├── _meta
│   ├── config.yml
│   ├── docs.asciidoc
│   └── fields.yml
└── firewall
    ├── _meta
    │   └── fields.yml
    ├── config
    │   ├── input.yml
    │   ├── liblogparser.js
    │   └── pipeline.js
    ├── ingest
    │   └── pipeline.yml
    ├── manifest.yml
    └── test
        ├── generated.log
        └── generated.log-expected.json
```

The `README.md`, `config.yml`, `docs.asciidoc` and `fields.yml` are the same, basically replacing the module/vendor/product name.

Same for `input.yml` and `manifest.yml`. They define the same variables and inputs (`tcp`, `udp` and `file`).

The `generated.log` files are autogenerated by the same program that converts the original XML files to Javascript. Generating logs using parser's patterns (some with overlap) and user-defined field names is hard. Some generated logs make more sense than others.

A few selected modules contain real logs that we were able to obtain from other sources. Currently:
- squid
- zscaler

The `liblogparser.js` is the helper Javascript library for the parser. It's important to review this file. It's the same for all filesets.

The `pipeline.js` is the autogenerated pipeline. Contains all the parsers and actions defined in the source XML. We're aware that some of these parsers are outdated and some partly broken regarding extra whitespace in patterns. There will be an ongoing effort to fix them.

A couple of modules already existed: 
- cisco (new fileset nexus)
- fortinet (new fileset clientendpoint)

In this case the `_meta/config.yml` and `_meta/docs.asciidoc` have been merged automatically. 

----

Some parsers are currently broken (have `tags`: `dissect_parsing_error`). I'm working on fixing those.

----

The code for the generator is in https://github.com/adriansr/nwdevice2filebeat

elasticmachine commented:
Pinging @elastic/siem (Team:SIEM)

elasticmachine commented:
## :broken_heart: Build Failed


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-19713/detail/PR-19713/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-19713/detail/PR-19713/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-19713/detail/PR-19713/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-19713/detail/PR-19713/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_19713.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #19713 updated]`

* Start Time: 2020-07-07T18:32:27.597+0000

* Duration: 9 min 26 sec





</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->


### Steps errors

  <details><summary>Expand to view the steps failures</summary>
  <p>
  
  * **Name**: `Make check`
    * Description: make check
    
    * Duration: 2 min 23 sec
    
    * Start Time: 2020-07-07T18:38:56.815+0000
    
    * [log](https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats-beats-mbp/pipelines/PR-19713/runs/1/steps/329/log)
    
  
  </p>
  </details>



<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-07-07T18:41:26.376Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-07T18:41:26.382Z] Stage "Kubernetes" skipped due to earlier failure(s)
[2020-07-07T18:41:26.606Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:26.616Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:26.626Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-07-07T18:41:26.636Z] Stage "Packetbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:26.645Z] Stage "dockerlogbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:26.654Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:26.663Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:26.672Z] Stage "Journalbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:26.682Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-07T18:41:27.811Z] Failed in branch Elastic Agent x-pack
[2020-07-07T18:41:27.819Z] Failed in branch Elastic Agent x-pack Windows
[2020-07-07T18:41:27.827Z] Failed in branch Elastic Agent Mac OS X
[2020-07-07T18:41:27.836Z] Failed in branch Filebeat oss
[2020-07-07T18:41:27.844Z] Failed in branch Filebeat x-pack
[2020-07-07T18:41:27.853Z] Failed in branch Filebeat Mac OS X
[2020-07-07T18:41:27.862Z] Failed in branch Filebeat x-pack Mac OS X
[2020-07-07T18:41:27.870Z] Failed in branch Filebeat Windows
[2020-07-07T18:41:27.879Z] Failed in branch Filebeat x-pack Windows
[2020-07-07T18:41:27.887Z] Failed in branch Auditbeat oss Linux
[2020-07-07T18:41:27.896Z] Failed in branch Auditbeat crosscompile
[2020-07-07T18:41:27.904Z] Failed in branch Auditbeat oss Mac OS X
[2020-07-07T18:41:27.913Z] Failed in branch Auditbeat oss Windows
[2020-07-07T18:41:27.922Z] Failed in branch Auditbeat x-pack
[2020-07-07T18:41:27.930Z] Failed in branch Auditbeat x-pack Mac OS X
[2020-07-07T18:41:27.939Z] Failed in branch Auditbeat x-pack Windows
[2020-07-07T18:41:27.948Z] Failed in branch Libbeat x-pack
[2020-07-07T18:41:27.956Z] Failed in branch Metricbeat OSS Unit tests
[2020-07-07T18:41:27.964Z] Failed in branch Metricbeat OSS Integration tests
[2020-07-07T18:41:27.972Z] Failed in branch Metricbeat Python integration tests
[2020-07-07T18:41:27.981Z] Failed in branch Metricbeat crosscompile
[2020-07-07T18:41:27.990Z] Failed in branch Metricbeat Mac OS X
[2020-07-07T18:41:27.998Z] Failed in branch Metricbeat x-pack Mac OS X
[2020-07-07T18:41:28.008Z] Failed in branch Metricbeat Windows
[2020-07-07T18:41:28.017Z] Failed in branch Metricbeat x-pack Windows
[2020-07-07T18:41:28.026Z] Failed in branch Winlogbeat Windows x-pack
[2020-07-07T18:41:28.035Z] Failed in branch Kubernetes
[2020-07-07T18:41:28.616Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:28.626Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:28.636Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-07-07T18:41:28.645Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:28.655Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:28.664Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-07T18:41:29.089Z] Failed in branch Packetbeat
[2020-07-07T18:41:29.098Z] Failed in branch dockerlogbeat
[2020-07-07T18:41:29.107Z] Failed in branch Journalbeat
[2020-07-07T18:41:29.547Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:29.557Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:29.566Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-07-07T18:41:29.575Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-07T18:41:29.678Z] Failed in branch Metricbeat x-pack
[2020-07-07T18:41:29.686Z] Failed in branch Winlogbeat
[2020-07-07T18:41:30.074Z] Failed in branch Heartbeat
[2020-07-07T18:41:30.083Z] Failed in branch Libbeat
[2020-07-07T18:41:30.091Z] Failed in branch Functionbeat
[2020-07-07T18:41:30.092Z] Stage "Generators" skipped due to earlier failure(s)
[2020-07-07T18:41:30.287Z] Failed in branch Generators
[2020-07-07T18:41:32.056Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-19713/src/github.com/elastic/beats
[2020-07-07T18:41:32.393Z] + find . -type f -name TEST*.xml -path */build/* -delete
[2020-07-07T18:41:32.416Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-19713/src/github.com/elastic/beats/Lint
[2020-07-07T18:41:32.860Z] + cat
[2020-07-07T18:41:32.861Z] + /usr/local/bin/runbld ./runbld-script
[2020-07-07T18:41:32.861Z] Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
[2020-07-07T18:41:39.458Z] runbld>>> runbld started
[2020-07-07T18:41:39.458Z] runbld>>> 1.6.12/f45d832f2ba0aa2722ab4ec1fda8ad140f027f8b
[2020-07-07T18:41:41.379Z] runbld>>> The following profiles matched the job 'Beats/beats-beats-mbp/PR-19713' in order of occurrence in the config (last value wins).
[2020-07-07T18:41:42.769Z] runbld>>> Debug logging enabled.
[2020-07-07T18:41:42.769Z] runbld>>> Storing result
[2020-07-07T18:41:43.032Z] runbld>>> Store result: created {:total 2, :successful 2, :failed 0} 1
[2020-07-07T18:41:43.032Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200707184142-5BF018DE
[2020-07-07T18:41:43.032Z] runbld>>> Adding system facts.
[2020-07-07T18:41:43.978Z] runbld>>> Adding vcs info for the latest commit:  61b1a84552b517fdbcb0d3d814661e1e8c289284
[2020-07-07T18:41:43.978Z] runbld>>> >>>>>>>>>>>> SCRIPT EXECUTION BEGIN >>>>>>>>>>>>
[2020-07-07T18:41:43.978Z] runbld>>> Adding /usr/lib/jvm/java-8-openjdk-amd64/bin to the path.
[2020-07-07T18:41:43.978Z] + echo 'Processing JUnit reports with runbld...'
[2020-07-07T18:41:43.978Z] Processing JUnit reports with runbld...
[2020-07-07T18:41:44.552Z] runbld>>> <<<<<<<<<<<< SCRIPT EXECUTION END <<<<<<<<<<<<
[2020-07-07T18:41:44.552Z] runbld>>> DURATION: 15ms
[2020-07-07T18:41:44.552Z] runbld>>> STDOUT: 40 bytes
[2020-07-07T18:41:44.552Z] runbld>>> STDERR: 49 bytes
[2020-07-07T18:41:44.552Z] runbld>>> WRAPPED PROCESS: SUCCESS (0)
[2020-07-07T18:41:44.552Z] runbld>>> Searching for build metadata in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-19713/src/github.com/elastic/beats
[2020-07-07T18:41:45.505Z] runbld>>> Storing build metadata: 
[2020-07-07T18:41:45.505Z] runbld>>> Adding test report.
[2020-07-07T18:41:45.505Z] runbld>>> Searching for junit test output files with the pattern: TEST-.*\.xml$ in: /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-19713/src/github.com/elastic/beats
[2020-07-07T18:41:46.079Z] runbld>>> Found 0 test output files
[2020-07-07T18:41:46.079Z] runbld>>> Test output logs contained: Errors: 0 Failures: 0 Tests: 0 Skipped: 0
[2020-07-07T18:41:46.340Z] runbld>>> Storing result
[2020-07-07T18:41:46.602Z] runbld>>> Store result: updated {:total 2, :successful 2, :failed 0} 2
[2020-07-07T18:41:46.602Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200707184142-5BF018DE
[2020-07-07T18:41:46.602Z] runbld>>> Email notification disabled by environment variable.
[2020-07-07T18:41:46.602Z] runbld>>> Slack notification disabled by environment variable.
[2020-07-07T18:41:52.453Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-19713
[2020-07-07T18:41:52.738Z] [INFO] getVaultSecret: Getting secrets
[2020-07-07T18:41:52.810Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-07-07T18:41:53.853Z] + chmod 755 generate-build-data.sh
[2020-07-07T18:41:53.853Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-19713/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-19713/runs/1 FAILURE 565985
[2020-07-07T18:41:53.853Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-19713/runs/1/steps/?limit=10000 -o steps-info.json
[2020-07-07T18:41:54.103Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-19713/runs/1/tests/?status=FAILED -o tests-errors.json

```

</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

andrewstucki commented on x-pack/filebeat/module/cylance/protect/test/generated.log-expected.json:
```diff
@@ -0,0 +1,3452 @@
+[
+    {
+        "@timestamp": "2016-01-29T08:09:59.000Z",
+        "event.category": "ZoneAdd",
```
We probably don't want to copy the `event.category` or `event.type` values directly from the vendor-specific ones since they won't usually align with the ECS-allowed values. In the meantime WDYT about moving all of the vendor-specific categorization to `event.action`?

leehinman replied:
+1 on not using vendor values in event.category.

adriansr replied:
Done. I'm mapping all to `event.action` (single value)

andrewstucki commented on x-pack/filebeat/module/cylance/protect/test/generated.log-expected.json:
Pretty sure this should just be a string. Also would it be possible to add a `related.user` array similar to how you did `related.ip`/

adriansr replied:
Done. Added a `related.user` that gets all the username fields, and one of the values will go to `user.name` (one of `username`,`c_username`,`uid`,`administrator`,`logon_id`,`owner`,`service_account` or `c_user_name`, left taking precedence over right).

andrewstucki commented on x-pack/filebeat/module/cylance/protect/test/generated.log-expected.json:
I'm assuming this is the network mac, could we add this under the `host.mac` field?

andrewstucki commented on x-pack/filebeat/module/cylance/protect/test/generated.log-expected.json:
should this be `source.ip` or `host.ip`?

adriansr replied:
The original parser is mapping this to the source address field. I'll keep it like that for now as I prefer not to modify the original parsers yet, but will keep a note to review this one.

andrewstucki commented on x-pack/filebeat/module/imperva/securesphere/config/liblogparser.js:
```diff
@@ -0,0 +1,2327 @@
+    "ec_outcome": {to:[{field: "event.outcome", setter: fld_set}]},
+    "event_description": {to:[{field: "message", setter: fld_prio, prio: 0}]},
+    "event_time": {convert: to_date, to:[{field: "@timestamp", setter: fld_set}]},
+    "event_type": {to:[{field: "event.category", setter: fld_set}]},
```
See comment from other modules, think we shouldn't set this with vendor-specific values. Maybe a better use in the `event.code` or `event.action` fields

adriansr replied:
done

andrewstucki commented on x-pack/filebeat/module/imperva/securesphere/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5693 @@
+            "imperva.securesphere",
+            "forwarded"
+        ],
+        "user.name": [
```
I'm thinking you probably just want to use the value from the original event that says `osUsername` here. The one for dbUsername could be added to `related.user` though. Also, it looks like the database schema name is getting added for some reason?

adriansr replied:
done. This fixed itself by selecting a single field for `user.name`

andrewstucki commented on x-pack/filebeat/module/imperva/securesphere/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5693 @@
+        "event.code": "Imperva",
+        "event.dataset": "imperva.securesphere",
+        "event.module": "imperva",
+        "event.original": "%IMPERVA-Imperva,dstIP=10.70.155.35,dstPort=892,dbUsername=tatno,srcIP=10.81.122.126,srcPort=4141,creatTime=29 January 2016 06:09:59,srvGroup=uam,service=untutl,appName=rad,event#=taliqu,eventType=Login,usrGroup=ommod,usrAuth=True,application=\"scivel\",osUsername=aqui,srcHost=radipis5408.mail.local,dbName=enatuse,schemaName=magn,bindVar=equuntu,sqlError=failure,respSiz ...
```
is `usrGroup` the group associated with the operating system user? If so, wondering if we can translate that to the ECS `group` fields

andrewstucki commented on x-pack/filebeat/module/fortinet/clientendpoint/test/generated.log-expected.json:
```diff
@@ -0,0 +1,2302 @@
+        "fileset.name": "clientendpoint",
+        "input.type": "log",
+        "log.flags": [
+            "dissect_parsing_error"
```
looks like pretty much all of these have parsing errors?

adriansr replied:
fixed

andrewstucki commented on x-pack/filebeat/module/infoblox/nios/test/generated.log-expected.json:
```diff
@@ -0,0 +1,2476 @@
+        "event.code": "cloud_api",
+        "event.dataset": "infoblox.nios",
+        "event.module": "infoblox",
+        "event.original": "Feb 12 13:12:33 com1060.api.example 10.14.94.160 cloud_api[tur]: proxying request to atio5608.www5.localhost(10.202.204.154) eFini https://www.example.org/exe/iatu.jpg?orsitame=reprehe#rsitam ggp issusci",
```
Doesn't look like we're capturing the first IP yet? Also, I'm wondering if it makes more sense to use the `source`/`destination` fields here rather than `host`?

adriansr replied:
The issue with this logs is that it captures the first IP in a field called hhostip, which we're not mapping because it's not documented, maybe I could append that to `host.ip` same as the `hostip` field.

andrewstucki replied:
So based off of the message and reading a bit on the cloud api stuff, it looks like the `hostip` field here (`10.202.204.154`) appears to be the IP that is receiving a request rather than the host that an event is occurring on or about some internals to the host (which is generally how I think about the `host` fields). I'm thinking that it probably makes sense more as either `destination` or `server` and maybe the `hhostip` is the originating server so `client` or `source`?

Either way, in the case of something like a proxy or load balancer i'd probably leave the `host.*` fields to represent the actual ip/mac/whatever other info of the proxy itself and fill in `source`/`destination` or `client`/`server` to describe the connections of the request. That way if someone wanted to see something like all the requests to a certain server from a given proxy they could do something like `host.name:myproxy.domain AND destination.ip:www.example.com`.

Is there a mechanism for generating some of these logs where we can figure out what the heck `hhostip` actually is?

andrewstucki commented on x-pack/filebeat/module/infoblox/nios/test/generated.log-expected.json:
```diff
@@ -0,0 +1,2476 @@
+        "rsa.internal.messageid": "ntpd",
+        "rsa.misc.event_source": "fugit7668.www5.invalid",
+        "rsa.time.day": "2",
+        "rsa.time.event_time": "2019-08-02T03:43:25.000Z",
```
this should probably be the value for `@timestamp` no?

adriansr replied:
The problem with this logs is that they don't contain a year, only day/month.

In that case I set the year to the current year (or the past year, if that would put the date more than 2 days in the future).

Due to this, I have to configure the tests to drop `@timestamp` for this fileset (in `test_modules.py`).

andrewstucki commented on x-pack/filebeat/module/zscaler/zia/test/generated.log-expected.json:
```diff
@@ -0,0 +1,7176 @@
+        "rsa.web.fqdn": "rci737.www5.example",
+        "service.type": "zscaler",
+        "source.bytes": 1884,
+        "source.ip": [
```
same as with squid. should we be using `client`/`server`?

leehinman commented on x-pack/filebeat/module/fortinet/clientendpoint/test/generated.log-expected.json:
```diff
@@ -0,0 +1,2302 @@
+        "event.code": "https",
+        "event.dataset": "fortinet.clientendpoint",
+        "event.module": "fortinet",
+        "event.original": "March 26 2016/03/26 aqui4726.mail.localhost proto=icmp service=https status=deny src=10.85.66.161 dst=10.131.115.96 src_port=2638 dst_port=1890 server_app=eum pid=654 app_name=rmagni traff_direct=sit block_count=5509 logon_user=onev@tenima1073.local msg=success",
```
should the ips, ports, proto & service be parsed out? seems odd that it isn't.

leehinman commented on x-pack/filebeat/module/imperva/securesphere/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5693 @@
+        "event.dataset": "imperva.securesphere",
+        "event.module": "imperva",
+        "event.original": "%IMPERVA-Imperva,dstIP=10.70.155.35,dstPort=892,dbUsername=tatno,srcIP=10.81.122.126,srcPort=4141,creatTime=29 January 2016 06:09:59,srvGroup=uam,service=untutl,appName=rad,event#=taliqu,eventType=Login,usrGroup=ommod,usrAuth=True,application=\"scivel\",osUsername=aqui,srcHost=radipis5408.mail.local,dbName=enatuse,schemaName=magn,bindVar=equuntu,sqlError=failure,respSiz ...
+        "event.outcome": "Success",
```
event.outcome should be lowercase "success"

leehinman commented on x-pack/filebeat/module/rapid7/nexpose/test/generated.log-expected.json:
```diff
@@ -0,0 +1,1901 @@
+        "observer.product": "Nexpose",
+        "observer.type": "Vulnerability",
+        "observer.vendor": "Rapid7",
+        "rsa.internal.messageid": "[Site:",
```
Is this right?, looks off with the [

adriansr replied:
There's something off in this parser or the log generator. I will check

leehinman commented on x-pack/filebeat/module/sonicwall/firewall/test/general.log-expected.json:
```diff
@@ -0,0 +1,639 @@
+        "fileset.name": "firewall",
+        "input.type": "log",
+        "log.flags": [
+            "dissect_parsing_error"
```
is the dissect failing?  seems odd we aren't getting src & dst fields out of this.

adriansr replied:
fixed

leehinman commented on x-pack/filebeat/module/squid/log/test/access1.log-expected.json:
```diff
@@ -0,0 +1,5610 @@
+        "rsa.network.domain": "login.yahoo.com",
+        "rsa.time.duration_time": 5006,
+        "rsa.time.event_time": "2006-09-08T04:21:52.000Z",
+        "rsa.time.event_time_str": "1157689312",
```
seems odd that the event_time_str field is populated with a number.

adriansr replied:
Yep, that's how this parser is defined. squid uses UNIX timestamps, which is captured it in event_time_str (raw). That is used to populate event_time and `@timestamp`.

leehinman commented on x-pack/filebeat/module/squid/log/test/access2.log-expected.json:
```diff
@@ -0,0 +1,5510 @@
+        "url.domain": "www.fas.harvard.edu",
+        "url.original": "http://www.fas.harvard.edu/~hpcws/nav/navbar_r3_c6.jpg",
+        "user.name": [
+            "-"
```
very minor.  I'm assuming "-" means no user name.  can we skip if "-"?

leehinman commented on x-pack/filebeat/module/tomcat/log/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5422 @@
+        "rsa.time.event_time": "2016-01-29T08:09:59.000Z",
+        "rsa.time.timezone": "OMST",
+        "rsa.web.alias_host": "https://example.com/illumqui/ventore.html?min=ite#utl",
+        "rsa.web.fqdn": "https://example.com/illumqui/ventore.html?min=ite#utl",
```
seems odd that these fields have full url

adriansr replied:
It's a bit odd. This parser is capturing the full URL and setting `fqdn` to that. There's a method to extract the FQDN from an URL, but it's not using it for some reason.

adriansr commented on x-pack/filebeat/module/sonicwall/firewall/test/general.log-expected.json:
```diff
@@ -0,0 +1,639 @@
+        "rsa.internal.messageid": "98",
+        "rsa.internal.msg": "Connection Opened",
+        "rsa.network.sinterface": "WAN",
+        "rsa.time.event_time": "2007-01-03T16:48:06.000Z",
```
fixed this parser.

adriansr commented on x-pack/filebeat/module/squid/log/test/access2.log-expected.json:
```diff
@@ -0,0 +1,5510 @@
+        "event.module": "squid",
+        "event.original": "1035368729.430    371 210.8.79.228 TCP_MISS/200 2136 GET http://www.fas.harvard.edu/~hpcws/nav/navbar_r3_c6.jpg - FIRST_PARENT_MISS/proxy2.syd.connect.com.au image/jpeg",
+        "fileset.name": "log",
+        "http.response.body.content": "navbar_r3_c6.jpg",
```
This is the field `webpage`, mapped to `http.response.body.content`. I don't think any parser is going to capture the full body of a request, so we might as well not map this field to ECS, as there is no field for the document name. Maybe `file.name`? @webmat WDYT?

adriansr commented on x-pack/filebeat/module/squid/log/test/access2.log-expected.json:
```diff
@@ -0,0 +1,5510 @@
+    {
+        "@timestamp": "2002-10-23T10:25:30.000Z",
+        "event.action": "TCP_MISS",
+        "event.code": "GET",
```
This will involve changing the original parser. I'll keep it in my notes for when we figure out how we want to maintain this in the future.

andrewstucki commented on x-pack/filebeat/module/fortinet/clientendpoint/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5702 @@
+        "event.dataset": "fortinet.clientendpoint",
+        "event.module": "fortinet",
+        "event.original": "January 29 06:09:59 boNemoe4402.www.invalid proto=udp service=http status=deny src=10.150.92.220 dst=10.102.123.34 src_port=7178 dst_port=3994 server_app=reeufugi pid=7880 app_name=enderitq traff_direct=sperna block_count=884 logon_user=billoi@oreetdol1714.internal.corp msg=failure",
+        "event.outcome": "Failure",
```
This should be lowercased, also what does `Failure` mean here? That a user/process was prevented from issuing an HTTP request?

adriansr replied:
I've mapped event.outcome to the expected values.

andrewstucki commented on x-pack/filebeat/module/fortinet/clientendpoint/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5702 @@
+        "event.dataset": "fortinet.clientendpoint",
+        "event.module": "fortinet",
+        "event.original": "April 24 00:25:25 reetdolo2770.www5.local proto=tcp service=pop3 status=deny src=10.12.44.169 dst=10.214.225.125 src_port=5710 dst_port=2121 server_app=inBCSedu pid=5722 app_name=tanimi traff_direct=nimadmin block_count=6499 logon_user=uam@temq1198.internal.example msg=success",
+        "event.outcome": "Failure",
```
this looks like it's always set to `Failure`? `msg:failure`, `msg:success`, and `msg:unknown` all appear to be `Failure`

adriansr replied:
This is something weird this parser does. It hardcodes `ec_outcome` to `Failure` (mapped to `event.outcome`). Then captures the `msg` in `result`, which is mapped to `rsa.misc.result`.

I'll see if that can be fixed by giving `result` priority over `ec_outcome` to populate `event.outcome`, but need to be careful as it might break other parsers.

andrewstucki commented on x-pack/filebeat/module/fortinet/clientendpoint/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5702 @@
+        "host.name": "tatn2376.www5.corp",
+        "input.type": "log",
+        "log.offset": 4299,
+        "network.direction": "ecillu",
```
is this just from munging the data or do we need to do some additional normalization? `ecillu` doesn't seem like a valid direction

adriansr replied:
This is the log generator not knowing how to populate direction fields. I'll update it.

adriansr commented:
@leehinman @andrewstucki I *think* I've addressed all the issues I could. Thanks a lot for your feedback, it's helping to add a lot of improvement.

ebeahan commented on x-pack/filebeat/module/squid/log/test/access1.log-expected.json:
```diff
@@ -0,0 +1,5610 @@
+        "rsa.time.event_time": "2006-09-08T04:21:52.000Z",
+        "rsa.time.event_time_str": "1157689312",
+        "rsa.web.alias_host": "login.yahoo.com",
+        "server.domain": "login.yahoo.com",
```
@andrewstucki Yes as you stated `client/server` is appropriate where there's a some type of session opened and each of the two systems have a clear role as either the `client` or `server` in the transaction. `source/destination` isn't limited to an unestablished or stateless connection; a `client/server` TCP session event could also populate `source/destination`.

In the case of a (assuming explicit) proxy, you would have two unique TCP connections established: one from host to proxy and another from proxy to origin server. Squid doesn't appear to log the events this way though (consolidating the two separate sessions into a single event in the logs). Since what is logged is the source host and the destination origin from the observer (squid), I think `source/destination` is appropriate in this case.

andrewstucki commented on x-pack/filebeat/module/squid/log/test/access2.log-expected.json:
```diff
@@ -0,0 +1,5610 @@
+        "event.module": "squid",
+        "event.original": "1035368730.746    297 210.8.79.228 TCP_MISS/200 1467 GET http://www.fas.harvard.edu/~hpcws/nav/navbar_r4_c1.jpg - FIRST_PARENT_MISS/proxy1.syd.connect.com.au image/jpeg",
+        "fileset.name": "log",
+        "http.response.body.content": "navbar_r4_c1.jpg",
```
thinking we might want to just drop this field since it's not aligning with ECS

adriansr replied:
This is using the field `webpage` which is defined as "captures the web page information". All the parsers that I've seen using it seem to do it to capture the name of the html document, so I think for now it's safe to map it to `file.name`, which is what we've done in other similar modules in the past.

WDYT?

andrewstucki replied:
I think I'd be fine with `file.name`, but I'm not entirely sure if the semantics of that are fine given:

1. this doesn't necessarily have the same information as normally associated with a filesystem entry, and
2. I'm assuming that in the case of url without a file prefix this might return something like `19713` from a url like `https://github.com/elastic/beats/pull/19713`?

That said, if @ebeahan or @webmat are fine with this being `file.name` I'm fine with that too, but I'd defer to them.

ebeahan replied:
I'm fine with `file.name` in this case, but it does appear we will run into instances of the caveat in the second bullet: https://github.com/elastic/beats/blob/4f80f24fa8b6b8631e54e8e681a7cff6d78b74cb/x-pack/filebeat/module/squid/log/test/access2.log-expected.json#L1800

andrewstucki commented on x-pack/filebeat/module/tomcat/log/test/generated.log-expected.json:
```diff
@@ -0,0 +1,5522 @@
+        "event.timezone": "OMST",
+        "fileset.name": "log",
+        "http.request.referrer": "https://mail.example.net/turadipi/aeca.htm?ntium=psaq#cer",
+        "http.response.body.content": "vol",
```
I'm assuming this also isn't the actual http content, so I think we should just drop it

adriansr commented on x-pack/filebeat/module/squid/log/test/access2.log-expected.json:
```diff
@@ -0,0 +1,5610 @@
+            "210.8.79.228"
+        ],
+        "related.user": [
+            "-"
```
I'm more inclined to leave this one as is for now, until a mechanism is added to make this kind of conditionals per device instead of adding code to all the pipelines for this specific case

adriansr commented on x-pack/filebeat/module/microsoft/dhcp/test/generated.log-expected.json:
```diff
@@ -0,0 +1,3042 @@
+        "source.ip": [
+            "10.103.246.190"
+        ],
+        "source.mac": "01:00:5e:8b:ba:06,ect,reetdolo,nrepreh,obeataev,lor",
```
fixed

adriansr commented:
Merged using admin privileges due to CI flakiness. All tests passing in my computer.
